tests/test_vpc.py

```diff
@@ -559,6 +562,133 @@ def test_interface_subnet(self):
+    def create_ec2_instance(self, ec2, size='t2.nano', ami_id='ami-43a15f3e'):
+        result = ec2.run_instances(ImageId=ami_id,
+                                   InstanceType=size,
+                                   MaxCount=1,
+                                   MinCount=1,
+                                   DisableApiTermination=False,
+                                   )
+        self.assertEquals(len(result['Instances']), 1)
+        self.addCleanup(ec2.terminate_instances, InstanceIds=[i['InstanceId'] for i in result['Instances']])
+
+        instance_data = result['Instances'][0]
+        while instance_data['State']['Code'] != 16:
+            time.sleep(5)
+            result = ec2.describe_instances(InstanceIds=(instance_data['InstanceId'],))
+            self.assertEquals(len(result['Reservations'][0]['Instances']), 1)
+            instance_data = result['Reservations'][0]['Instances'][0]
+
+        return instance_data
+
+    def allocate_network_address(self, ec2, domain_type):
+        network_addr = ec2.allocate_address(Domain=domain_type)
+        self.addCleanup(self.release_if_still_present, ec2, network_addr)
+        return network_addr
+
+    def core_release_op(self, factory, ec2, network_addr, force=False):
+        alloc_id = network_addr['AllocationId']
+
+        p = self.load_policy({
+            'name': 'release-network-addr',
+            'resource': 'network-addr',
+            'filters': [
+                {'AllocationId': alloc_id},
+            ],
+            'actions': [{'type': 'release', 'force': force}], },
+            session_factory=factory)
+
+        resources = p.run()
+
+        self.assertEqual(len(resources), 1)
+        with self.assertRaises(BotoClientError) as e_cm:
+            ec2.describe_addresses(AllocationIds=[ alloc_id ])
+        e = e_cm.exception
+        self.assertEqual(e.response['Error']['Code'], 'InvalidAllocationID.NotFound')
+        self.assertIn(alloc_id, e.response['Error']['Message'])
+
+    @functional
+    def test_release_detached_vpc(self):
+        factory = self.replay_flight_data('test_release_detached_vpc')
+
+        session = factory()
+        ec2 = session.client('ec2')
+
+        network_addr = self.allocate_network_address(ec2, 'vpc')
+        self.core_release_op(factory, ec2, network_addr)
+
+    @functional
+    def test_release_detached_classic(self):
+        factory = self.replay_flight_data('test_release_detached_classic')
+
+        session = factory()
+        ec2 = session.client('ec2')
+
+        network_addr = self.allocate_network_address(ec2, 'classic')
+        self.core_release_op(factory, ec2, network_addr)
+
+    # This is not a functional test because spinning up the ec2 takes too long
+    def test_release_attached_ec2_vpc(self):
+        factory = self.replay_flight_data('test_release_attached_ec2_vpc')
+
+        session = factory()
+        ec2 = session.client('ec2')
+
+        ec2_instance = self.create_ec2_instance(ec2)
```

Comment:
we don't need to create an instance if we're not doing a functional test, we can just use an extant one.